extensions/podman/packages/extension/src/podman-machine-stream.ts

```diff
@@ -0,0 +1,141 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as fs from 'node:fs';
+
+import type {
+  Event,
+  ProviderConnectionShellAccess,
+  ProviderConnectionShellAccessData,
+  ProviderConnectionShellAccessError,
+  ProviderConnectionShellAccessSession,
+  ProviderConnectionShellDimensions,
+} from '@podman-desktop/api';
+import { EventEmitter } from '@podman-desktop/api';
+import type { ClientChannel } from 'ssh2';
+import { Client } from 'ssh2';
+
+import type { MachineInfo } from './extension';
+
+export class ProviderConnectionShellAccessImpl implements ProviderConnectionShellAccess {
```

Comment:
```suggestion
export class ProviderConnectionShellAccessImpl implements ProviderConnectionShellAccess extends Disposable {
```

Makes ProviderConnectionShellAccessImpl  extends Disposable, and add it to the subscriptions, so it is cleaned up on extension disactivate

Reply:
I would have to change the api.d.ts file for extending the `ProviderConnectionShellAccess`, I can add the `extends Disposable` before `implements` but then I get additional errors like "Class extends value #<Object> is not a constructor or null

Reply:
did you import Disposable class from the api ?

Reply:
Use implements would work **

see the example in ai-lab

https://github.com/containers/podman-desktop-extension-ai-lab/blob/8e9ce351a673be43e54a21c2cfb35555861de50d/packages/backend/src/managers/inference/inferenceManager.ts#L37